air/src/trace/chiplets/memory.rs

```diff
@@ -1,53 +1,62 @@
-pub const NUM_SELECTORS: usize = 2;
+// --- OPERATION SELECTORS ------------------------------------------------------------------------
 
-/// Type for Memory trace selectors.
-///
-/// These selectors are used to define which operation and memory state update (init & read / copy &
-/// read / write) is to be applied at a specific row of the memory execution trace.
-pub type Selectors = [Felt; NUM_SELECTORS];
+/// Specifies the value of the `READ_WRITE` column when the operation is a write.
+pub const MEMORY_WRITE: Felt = ZERO;
+/// Specifies the value of the `READ_WRITE` column when the operation is a read.
+pub const MEMORY_READ: Felt = ONE;
+/// Specifies the value of the `ELEMENT_OR_WORD` column when the operation is over an element.
+pub const MEMORY_ACCESS_ELEMENT: Felt = ZERO;
+/// Specifies the value of the `ELEMENT_OR_WORD` column when the operation is over a word.
+pub const MEMORY_ACCESS_WORD: Felt = ONE;
 
-// --- OPERATION SELECTORS ------------------------------------------------------------------------
+// --- BUS LABELS ------------------------------------------------------------------------
 
-/// Specifies an operation that initializes new memory and then reads it.
-pub const MEMORY_INIT_READ: Selectors = [ONE, ZERO];
+// All bus labels encode the chiplet selector (1, 1, 0), as well as the read/write and element/word
+// columns. The purpose of the label is to force the chiplet to assign the correct values to the
+// read/write and element/word columns. We also include the chiplet selector as a "namespace" for
+// memory chiplet labels (to really ensure they don't collide with labels from other chiplets).
 
-/// Specifies an operation that copies existing memory and then reads it.
-pub const MEMORY_COPY_READ: Selectors = [ONE, ONE];
+/// Unique label when r/w=0 and e/w=0.
+pub const MEMORY_WRITE_ELEMENT_LABEL: u8 = 0b11000;
 
-/// Specifies a memory write operation.
-pub const MEMORY_WRITE: Selectors = [ZERO, ZERO];
+/// Unique label when r/w=0 and e/w=1.
+pub const MEMORY_WRITE_WORD_LABEL: u8 = 0b11001;
 
-/// Unique label computed as 1 plus the full chiplet selector with the bits reversed.
-/// mem_read selector=[1, 1, 0, 1], rev(selector)=[1, 0, 1, 1], +1=[1, 1, 0, 0]
-pub const MEMORY_READ_LABEL: u8 = 0b1100;
+/// Unique label when r/w=1 and e/w=0.
+pub const MEMORY_READ_ELEMENT_LABEL: u8 = 0b11010;
 
-/// Unique label computed as 1 plus the full chiplet selector with the bits reversed.
-/// mem_write selector=[1, 1, 0, 0] rev(selector)=[0, 0, 1, 1] +1=[0, 1, 0, 0]
-pub const MEMORY_WRITE_LABEL: u8 = 0b0100;
+/// Unique label when r/w=1 and e/w=1.
+pub const MEMORY_READ_WORD_LABEL: u8 = 0b11011;
 
 // --- COLUMN ACCESSOR INDICES WITHIN THE CHIPLET -------------------------------------------------
 
-/// The number of elements accessible in one read or write memory access.
-pub const NUM_ELEMENTS: usize = 4;
-
+/// Column to hold the whether the operation is a read or write.
+pub const READ_WRITE_COL_IDX: usize = 0;
```

Comment:
nit: should we name this `READ_OR_WRITE` to be consistent with `ELEMENT_OR_WORD`?